### DISCUSSION TITLE
multi-select-checkboxes

### DISCUSSION THREAD

**Discussion by janlin002** [JedWatson/react-select#5620] [Q&A] [answered]
How can I achieve this effect using react-select?
(first checkbox is select all)
And if the number of selected items is more than 5, the actual content is not displayed.
<img width="372" alt="截圖 2023-04-27 上午11 31 18" src="https://user-images.githubusercontent.com/68535430/234752797-64e2a144-f033-4ef1-9744-8712cfd0c163.png">
![截圖 2023-04-26 下午6 02 49](https://user-images.githubusercontent.com/68535430/234752320-d619d441-c640-4f66-8a0f-78ec99635826.png)

**Comment by manjushsh** [comment]
Hi @janlin002, for showing checkbox and 'Select all' you can achieve with custom components. [Example Sandbox](https://codesandbox.io/s/multiselect-with-checkbox-gsg1o3) or [this thread](https://github.com/JedWatson/react-select/issues/3543) or [Stack Overflow](https://stackoverflow.com/a/61250357)
For three state checkbox there are[ package you can use](https://www.npmjs.com/package/react-three-state-checkbox) or refer some online articles.
Hope this helps

**Comment by janlin002** [comment]
Thank you for your help, but I would like it to automatically collapse when it exceeds the input height, instead of increasing the input height.
<img width="372" alt="截圖 2023-04-27 上午11 31 18" src="https://user-images.githubusercontent.com/68535430/235034125-216d9a8a-e7ef-4dbb-bd4a-f7861545cab8.png">

**Reply by manjushsh** [reply, marked as answer]
Hi @janlin002, I have [updated sandbox with change](https://codesandbox.io/s/multiselect-with-checkbox-gsg1o3?file=/src/App.js).
On clicking +n blocks, all extra blocks, all those extra blocks will be removed.
You can customize it with css to get desired output

**Reply by janlin002** [reply]
Thank you for your help, it has been very helpful to me.

**Reply by manjushsh** [reply]
I'm glad it helped

**Reply by User6531** [reply]
Hello @manjushsh, that's great, thanks for your example. Could you also assist me? I need to add a checkbox through props, for example, `optionHasCheckmark`, or add it when it's in `multiselect` mode. Do you have any ideas on how to implement this?

eg:
```tsx
const CustomOption = <Option,>(optionProps: OptionProps<Option>) => {
  const { innerProps, children, isSelected } = optionProps;

  return (
    <S.CustomStyledOption
      {...innerProps}
      ref={innerProps.ref as React.RefObject<HTMLDivElement>}
    >
      {optionHasCheckmark && <Checkbox checked={isSelected} />}
      {children}
    </S.CustomStyledOption>
  );
};
```

**Reply by User6531** [reply]
I found solution

```tsx
const CustomOption = <Option,>(optionProps: OptionProps<Option>) => {
  const {
    innerProps,
    children,
    isSelected,
    selectProps: { optionHasCheckmark },
  } = optionProps;

  return (
    <S.CustomStyledOption
      {...innerProps}
      ref={innerProps.ref as React.RefObject<HTMLDivElement>}
    >
      {optionHasCheckmark && <Checkbox checked={isSelected} />}
      {children}
    </S.CustomStyledOption>
  );
};
```
